src/Pipeline.cc

```diff
@@ -58,6 +67,10 @@ Pipeline::popMe(const Http::StreamPointer &which)
     // in reality there may be multiple contexts doing processing in parallel.
     // XXX: pipeline still assumes HTTP/1 FIFO semantics are obeyed.
     assert(which == requests.front());
+    if (which->mayUseConnection()) {
+        Assure(requests.size() == 1);
```

Comment:
This assertion might be (essentially accidentally) true in current PR code but it is not based on any HTTP logic AFAICT: The last transaction may remove itself from the pipeline _regardless_ of how many other transactions are currently in the pipeline. That applies to all last transactions, including transactions that mayUseConnection(). Why do we need to assert this condition here?

Reply:
> The last transaction may remove itself from the pipeline

How can this happen, i.e., the last transaction (or any other arbitrary one) remove itself from the pipeline? We do not have API for this. What we have is popMe(), removing the front element (transaction) from the queue. Since we do not add() until front() is still using connection, only back() element may use connection. That is what this assertion checks.